magento/magento2
# ISSUE-26481 - Allow 'Is Default' radio to be cleared when editing swatch and dropdown attributes

### Description (*)
Partially fixes problems outlined in issue [26481](https://github.com/magento/magento2/issues/26481). Currently, there is no way to deselect an Is Default selection when editing swatch and dropdown attributes in admin.

![Screen Shot 2020-04-07 at 9 35 24 AM](https://user-images.githubusercontent.com/3081921/78682044-21d04d80-78b3-11ea-8b3a-0bb41df9b0c8.png)

This PR makes it possible to click the selected radio again to deselect it, effectively removing the default value in the eav_attribute table on save. As outlined in issue [26481](https://github.com/magento/magento2/issues/26481), there are a number of reasons that an admin user may wish to undo the default on an attribute.

Note that this PR only adds the ability to remove the default selection on the attribute itself. Existing product data for the attribute is unaffected. Other challenges mentioned in [26481](https://github.com/magento/magento2/issues/26481) (such as default values being applied to products at unintended times) are not addressed by this PR as the exact expectation for the behavior of a default value is debatable.

### Fixed Issues (if relevant)
1. Fixes #26481: Unable to remove default text swatch value and value automatically assigning

### Manual testing scenarios (*)
1. In admin, go to Stores > Attributes / Product.
2. Click Add New Attribute.
3. Set "Catalog Input Type for Store Owner" to Dropdown.
4. Under Manage Options, click Add Option to add multiple options.
5. Click the "Is Default" radio on one of the options, complete any other required fields, and click Save and Continue Edit.
6. On page reload, click the currently selected "Is Default" radio. It should become unchecked.
7. Save the attribute again. On page reload, all "Is Default" radios should still be unchecked.
8. Repeat the steps above using Visual Swatch or Text Swatch as the "Catalog Input Type for Store Owner".

### Contribution checklist (*)
